CMakeLists.txt

```diff
@@ -306,13 +306,18 @@ endif()
 
 find_package(Perl)
 
-option(BUILD_DOCS "to build manual pages" ON)
-option(ENABLE_MANUAL "to provide the built-in manual" OFF)
-
-if(ENABLE_MANUAL AND PERL_FOUND)
-  curl_nroff_check()
-  if(NROFF_USEFUL)
-    set(USE_MANUAL ON)
+option(BUILD_DOCS "to build libcurl man pages" ON)
+option(ENABLE_MANUAL "to provide the built-in manual and man page for curl tool" OFF)
+
+if(ENABLE_MANUAL OR BUILD_DOCS)
+  if(PERL_FOUND)
+    curl_nroff_check()
+    if(NROFF_USEFUL)
+      set(HAVE_MANUAL_TOOLS ON)
```

Comment:
```suggestion
      set(HAVE_MANUAL_TOOLS ON)
      set(USE_MANUAL ON)
```

Reply:
This was supposed to be part of my proposal below

Reply:
`USE_MANUAL` isn't use any longer by CMake after this patch. The necessary C macro is defined in `src/CMakeList.txt`, so the above is unnecessary unless I missed something.

Reply:
Updated the commit message to reflect that.

Reply:
See my proposal below, where I suggest restoring its definition in curl_config.h.

Reply:
Why do you think that would be better?

It'd be two lines vs. one, and a `set(USE_MANUAL ON)` doesn't
readily conveys that we mean this for `curl_config.h`, and only
for that.

Reply:
Most of all, since the `USE_MANUAL` macro is checked in many places in the source, it's kinda odd not to have it being part of the collection of other `USE_` macros.

Reply:
It's used in 4 places, all inside `src`. Defining it via CMake and `lib/curl_config.h.cmake` makes the logic more difficult to follow IMO. It would also allow to set this accidentally and breaking the build in weird ways. Esp considering that `cmake -DUSE_MANUAL=ON` was a workaround for the broken `-DENABLE_MANUAL=ON` option for all previous releases. Also having the C-level macro set for `lib` is unncessary and a source for confusion.

Reply:
You know, I'm stepping down from this...  it's starting to become a bike shed, if you know what I mean, and not really that important.

Reply:
I do, no worries and thanks for your feedback!